common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleClient.java

```diff
@@ -97,15 +97,16 @@ public void fetchBlocks(
       String execId,
       String[] blockIds,
       BlockFetchingListener listener,
-      DownloadFileManager downloadFileManager) {
+      DownloadFileManager downloadFileManager,
+      int shuffleGenerationId) {
```

Comment:
nit: seems better to put `shuffleGenerationId` parameter right after `String execId`

Reply:
Copy that, done in 96bab1e.